log4j-api/src/main/java/org/apache/logging/log4j/CloseableThreadContext.java

```diff
@@ -206,7 +207,7 @@ public void close() {
         }
 
         private void closeMap() {
-            final Map<String, String> valuesToReplace = new HashMap<>(originalValues.size());
+            final Map<String, String> valuesToReplace = Maps.newHashMap(originalValues.size());
```

Comment:
I've reviewed all `log4j-api` and `log4j-core` changes, and found this to be only change that one _might_ claim to be of importance. I prefer a PR that

1. Replaces all `HashMap#new(int)` calls with a `HashMap#new()` (Since they're all incorrectly implemented, it is better to just stick to defaults.)
2. Replaces only this line in `CloseableThreadContext` with `new HashMap((int) Math.ceil(originalValues.size() / 0.75))`

@ppkarwasz, WDYT?

Reply:
Can you point out an example, where the usage of `HashMap#new(int)` is inappropriate?

I am mostly +1 on the proposed changes, as long as they don't introduce a new API. We have already dealt with methods and classes backported from newer JDK versions.

Reply:
> All changes this PR highlights show `HashMap#new(int)` usages where the author of the associated piece of code _incorrectly_ assumed the provided capacity would avoid allocations, but it does not.

I must be tired, but could you point me to some examples? I checked a couple of usages and, if the capacity of the map is computed correctly, they don't seem to cause the map to be resized.

Reply:
For one, all `*Test` suggestions are redundant. The rest of the usages are incorrect, see `ThreadDumpMessage`:

```
Map<ThreadInformation, StackTraceElement[]> threads = new HashMap<>(map.size());
for (Map.Entry<Thread, StackTraceElement[]> entry : map.entrySet()) {
    threads.put(...);
}
```

Author had the intention of creating an initial `HashMap` of capacity `map.size()` to avoid resizing.
1. This is _incorrect_, since it _will be_ resized at the .75 mark
2. This code is hardly on a hot path, if used at all

I'm not saying what this PR implements is _incorrect_, instead, what is already there in the existing code base is incorrect. Plus, there is nothing to be corrected if just the defaults are used. Being extra smart does not buy anything but complicate the code.

Reply:
> * Keep the current code **unchanged**, accepting that it may trigger a resize for some values of n (but not all, since the threshold is always 0.75 × the smallest power of 2 ≥ n).

That is a fair path forward. This effectively translates to only fixing the `CloseableThreadContext` with an inline math operation, @ppkarwasz, do you agree?

Reply:
The `internal` packages are not for users, but for us: BND Baseline will not complain if we remove the class and we'll not address any users complains, since they have been sufficiently warned.

I am for merging this one, if my comment above is addressed.

Reply:
I've made the changes suggested by @ppkarwasz, thanks for the discussion / review.